XpringKit/XRPCurrency.swift

```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+/// A currency on the XRP Ledger
+/// - SeeAlso: 
+public struct XRPCurrency: Equatable {
+  public let name: String
+  public let code: Data
+}
```

Comment:
What does the `Equatable` keyword mean here? Why do we need this `struct` and also the `internal extension`?

Reply:
`struct` is a class in swift, except it has a bunch of useful properties, like copy on assignment and a bunch of generated code under the hood (ie. for equality). Basically, they're like light classes. 
https://docs.swift.org/swift-book/LanguageGuide/ClassesAndStructures.html

`: Equatable` Means that we're conforming to the `Equatable` protocol. This basically means we implement an `equals` method, and then we can do things like `XCTAssertEquals(someCurrencyStruct, someOtherCurrencyStruct)`. `XCTAssertEquals` takes in two `Equatable` objects as opposed to an exact type:
https://developer.apple.com/documentation/swift/equatable

If we used a class we'd have to write our own equals method, but because we're using a `struct` and all of our iVars are also `equatable` the swift compiler automatically synthesizes the methods to make this class conform to `equatable` at runtime.

Reply:
I should have filled in the `SeeAlso` above. 

> The rippled APIs support two formats of currency code for issued currencies:
> - Standard Currency Codes: As a 3-character string such as "EUR" or "USD".
> - Nonstandard Currency Codes: As a 160-bit hexadecimal string, such as "0158415500000000C1F76FF6ECB0BAC600000000". This is uncommon.


https://xrpl.org/currency-formats.html#currency-codes